lang/java/reef-io/src/test/java/org/apache/reef/io/network/NetworkConnectionServiceTest.java

```diff
@@ -361,23 +345,11 @@ public void testMessagingNetworkConnServiceBatchingRate() throws Exception {
         messagingTestService.registerTestConnectionFactory(groupCommClientId, numMessages, monitor, codec);
         try (final Connection<String> conn =
                  messagingTestService.getConnectionFromSenderToReceiver(groupCommClientId)) {
-
-          // build the message
-          final StringBuilder msb = new StringBuilder();
-          for (int i = 0; i < size; i++) {
-            msb.append("1");
-          }
-          final String message = msb.toString();
-
           final long start = System.currentTimeMillis();
           try {
+            conn.open();
             for (int i = 0; i < numMessages; i++) {
-              final StringBuilder sb = new StringBuilder();
-              for (int j = 0; j < batchSize / size; j++) {
-                sb.append(message);
-              }
-              conn.open();
-              conn.write(sb.toString());
+              conn.write(message);
```

Comment:
I think this change doesn't preserve functionality: previously we've written `numMessages * batchSize / size` messages, now we write only `numMessages`. Should keep `batchSize / size` multiplier